### PR TITLE
Fix fast-path cache

### PR DESCRIPTION
**Pull request by sharwell**
For default **.editorconfig** scenarios, `SymbolNamesWithValueOption<TValue>.Empty` is used. This has an empty set of wildcard match names, but still caches symbols. Since the `Empty` value is static, this means the process keeps a cached copy of _all_ symbols for which `Contains` is called until the process runs out of memory.

This issue is also the reason why other `ConditionalWeakTable<TKey, TValue>` instances were never releasing values, since the `Compilation` keys were retaining strong roots through this path.

Fixes [AB#1242125](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1242125)

### CODE REVIEW

File: src/Utilities/Compiler/Options/SymbolNamesWithValueOption.cs [hunk 1]
```diff
@@ -277,7 +275,6 @@ private bool TryGetFirstWildcardMatch(ISymbol symbol, out KeyValuePair<string, T
             if (_wildcardNamesBySymbolKind.IsEmpty)
             {
                 firstMatch = NoWildcardMatch;
-                _wildcardMatchResult.AddOrUpdate(symbol, firstMatch, (s, match) => NoWildcardMatch);
```
Comment (sharwell) [inline src/Utilities/Compiler/Options/SymbolNamesWithValueOption.cs:280]: 📝 This is the bug fix. The rest is code cleanup.

**Review by mavasani — APPROVED** [review]
LGTM, but I would certainly want @Evangelink's review to confirm it does not regress perf on the path where the user has actually specified this option.

**Comment by sharwell** [comment]
> does not regress perf

`_wildcardNamesBySymbolKind` is immutable, so if it is empty for one call through this path it will be empty for all subsequent calls through this instance. This is an earlier return than the first point where this cache is ever used.

**Comment by mavasani** [comment]
@sharwell I can release new 3.3.2 analyzer NuGet packages once this PR is merged in.

**Pull request merged**